concept/type_/type_cache.rs

```diff
@@ -176,7 +171,10 @@ impl TypeCache {
         })
     }
 
-    fn build_common_cache<Snapshot: ReadableSnapshot, T: ReadableType<'static, 'static, SelfRead = T>>(snapshot: &Snapshot, type_: T) -> CommonTypeCache<T::SelfRead> {
+    fn build_common_cache<Snapshot: ReadableSnapshot, T>(snapshot: &Snapshot, type_: T) -> CommonTypeCache<T>
+    where
+        T: TypeAPI<'static> + TypeAPITraits + ReadableType<Output<'static>=T>,
```

Comment:
I thin kwe should either stick with the `impl` shorthand when we don't need to name the type:
```suggestion
    fn build_common_cache<T>(snapshot: &impl ReadableSnapshot, type_: T) -> CommonTypeCache<T>
    where
        T: TypeAPI<'static> + TypeAPITraits + ReadableType<Output<'static>=T>,
```
or move all bounds to the `where` clause:
```suggestion
    fn build_common_cache<Snapshot, T>(snapshot: &Snapshot, type_: T) -> CommonTypeCache<T>
    where
        Snapshot: ReadableSnapshot,
        T: TypeAPI<'static> + TypeAPITraits + ReadableType<Output<'static>=T>,
```
Mixing inline bounds and `where` isn't very nice to read